modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/odata/entity/v1_0/ObjectEntryEntityModel.java

```diff
@@ -189,17 +190,24 @@ private List<EntityField> _getRelatedObjectDefinitionEntityFields(
 		List<EntityField> relatedObjectDefinitionEntityFields = new ArrayList<>(
 			relatedObjectDefinitionEntityFieldsMap.values());
 
+		if (_objectDefinitionsHandled.contains(
+				relatedObjectDefinition.getObjectDefinitionId())) {
+
+			_objectDefinitionsHandled.remove(
+				objectDefinition.getObjectDefinitionId());
+
+			return relatedObjectDefinitionEntityFields;
+		}
+
 		List<ObjectRelationship> relatedObjectDefinitionObjectRelationships =
 			ObjectRelationshipLocalServiceUtil.getAllObjectRelationships(
 				relatedObjectDefinition.getObjectDefinitionId());
 
 		for (ObjectRelationship relatedObjectRelationship :
 				relatedObjectDefinitionObjectRelationships) {
 
-			if ((relatedObjectRelationship.getObjectRelationshipId() ==
-					objectRelationship.getObjectRelationshipId()) ||
-				_relatedObjectRelationships.contains(
-					relatedObjectRelationship.getObjectRelationshipId())) {
+			if (relatedObjectRelationship.getObjectRelationshipId() ==
+					objectRelationship.getObjectRelationshipId()) {
```

Comment:
As we discussed, here we should avoid creating the field when the new related objectDefinition has been handled before

Reply:
Done